src/components/Reports/Reports.js

```diff
@@ -142,37 +139,37 @@ const Reports = ({ dependencies: { datahubClient } }) => {
                     today={state.dailyView}
                     emailFilter={'with_email'}
                     visits={totalVisits.withEmail}
-                    loading={totalVisits.loading}
+                    loading={state.loading || totalVisits.loading}
                   />
                 </div>
-                {!state.dailyView ? (
+                {!state.dailyView && (
                   <div className="col-sm-12 m-b-24">
                     <ReportsDailyVisits
-                      domainName={state.domainSelected.name}
+                      domainName={state.domainSelected?.name}
```

Comment:
Is it a possible scenario? I mean is rendering this element fine when the _state.domainSelected_ does not have value?

Reply:
yes, it's possible. Because the `<ReportsDailyVisits />` is shown when the domains are being searched, in that case the `state.domainSelected` is `null`

Reply:
Thanks for the clarification! now `ReportsDailyVisits` is ready to deal with `domainName = null` and before it did not broke because these elements were not being rendered when loading was true.

This is a central part of your change, not only a refactoring.